.travis.yml

```diff
@@ -16,21 +21,39 @@ before_install:
   - 'npm config set registry http://registry.npmjs.org/'
   - 'export DISPLAY=:99.0'
   - 'sh -e /etc/init.d/xvfb start'
+  - 'sudo add-apt-repository -y ppa:webupd8team/java'
+  - 'sudo apt-get update'
+  - 'sudo apt-get install -y libappindicator1 fonts-liberation'
+  - 'sudo apt-get install -y oracle-java8-installer'
+  - 'sudo apt install -y oracle-java8-set-default'
+  - 'wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb'
+  - 'sudo dpkg -i google-chrome*.deb'
+  - 'npm install -g cypress-cli' #test using cypress cli for integrationg testing
+  - 'wget https://seqwaremaven.oicr.on.ca/artifactory/collab-release/io/dockstore/dockstore-webservice/1.1-rc.2/dockstore-webservice-1.1-rc.2.jar'
+  - 'chmod u+x dockstore-webservice-1.1-rc.2.jar'
   - 'sleep 3' # https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI
 install:
   - 'npm install --global grunt-cli bower yo generator-karma generator-angular'
-  - 'npm install karma jasmine-core'
   - 'sudo apt-get install -y ruby ruby-dev make'
   - 'gem install compass'
   - 'npm update'
   - 'bower update'
-  - 'node_modules/protractor/bin/webdriver-manager update --standalone --firefox'
-  - 'sleep 5'
-  - 'node_modules/protractor/bin/webdriver-manager start --standalone --firefox &'
-  - 'sleep 5'
-after_script:
-- cat ./build/coverage/**/lcov.info | ./node_modules/coveralls/bin/coveralls.js
-after_success: grunt coveralls
+
+before_script:
+  - grunt # this does syntax checking, unit testing, and jshint
+  - grunt serve --silent & #make sure it runs the server first before testing
+  - 'sleep 10'
```

Comment:
Make a ticket to use dockerize or similar here, this looks fragile depending on performance